tools/topology/topology1/development/sof-imx8-compr-wm8960.m4

```diff
@@ -0,0 +1,96 @@
+#
+# Topology with codec_adapter processing component for i.MX8QM/i.MX8QXP
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`sai.m4')
+include(`pcm.m4')
+include(`buffer.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include DSP configuration
+include(`platform/imx/imx8.m4')
+
+
+# Post process setup config
+
+ #codec Post Process setup config
+#
+# Define the pipelines
+#
+# PCM0 <----> volume <-----> SAI3 (wm8960)
+#
+
+
+DECLARE_SOF_RT_UUID("Cadence Codec", cadence_codec_uuid, 0xd8218443, 0x5ff3,
+                    0x4a4c, 0xb3, 0x88, 0x6c, 0xfe, 0x07, 0xb9, 0x56, 0xaa);
+
+define(`CA_UUID', cadence_codec_uuid)
+
+# Post process setup config
+define(`CA_SETUP_CONTROLBYTES',
+``	bytes "0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,'
+`	0x24,0x00,0x00,0x00,0x00,0x10,0x00,0x03,'
+`	0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`	0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`	0x07,0x20,0x00,0x80,0x0C,0x00,0x00,0x00,'
+`	0x18,0x00,0x00,0x00,0x03,0x20,0x00,0x80,'
+`	0x0C,0x00,0x00,0x00,0x02,0x00,0x00,0x00,'
+`	0x00,0x10,0x00,0x80,0x0C,0x00,0x00,0x00,'
+`	0x18,0x00,0x00,0x00"'')
+
+define(`CA_SETUP_CONTROLBYTES_MAX', 300)
+
+undefine(`DAI_PERIODS')
+define(`DAI_PERIODS', 8)
```

Comment:
I think it might work with 2 or 3 as usual, i remember that I added this in the beginning to compensate for the non-symetric sizes of codec_adapter input and output.

Eg. the decoder will output a data much bigger at the output after decompression. I could revise this at some point. But this proved that just works. 

We have lots of memory for buffers so it was never a concern.

Reply:
I am afraid this hides a scheduling problem. In some cases a frame will generate double-digit ms, and it's nearly impossible to complete the decoding for one frame in 1ms. The point is that decoding should not be a low-latency task and that there should be enough intermediate buffering between the decoder and the dai. The DAI period should not depend on what's upstream.